native/cocos/renderer/pipeline/xr/ar/ARStage.cpp

```diff
@@ -0,0 +1,145 @@
+        framegraph::TextureHandle depth;
+    };
+    auto *const sceneData = _pipeline->getPipelineSceneData();
+
+    float shadingScale{sceneData->getShadingScale()};
+
+    auto arSetup = [&](framegraph::PassNodeBuilder &builder, RenderData &data) {
+        gfx::TextureInfo colorTexInfo = {
+            gfx::TextureType::TEX2D,
+            gfx::TextureUsageBit::COLOR_ATTACHMENT | gfx::TextureUsageBit::SAMPLED,
+            gfx::Format::RGBA16F,
+            static_cast<uint>(camera->getWindow()->getWidth() * shadingScale),
+            static_cast<uint>(camera->getWindow()->getHeight() * shadingScale),
+        };
+        data.outputTex = builder.create(RenderPipeline::fgStrHandleOutColorTexture, colorTexInfo);
+
+        framegraph::RenderTargetAttachment::Descriptor colorAttachmentInfo;
+        colorAttachmentInfo.usage = framegraph::RenderTargetAttachment::Usage::COLOR;
+        colorAttachmentInfo.loadOp = gfx::LoadOp::CLEAR;
+
+        colorAttachmentInfo.endAccesses = {gfx::AccessFlagBit::COLOR_ATTACHMENT_WRITE};
+
+        data.outputTex = builder.write(data.outputTex, colorAttachmentInfo);
+
+        // depth
+        gfx::TextureInfo depthTexInfo{
+            gfx::TextureType::TEX2D,
+            gfx::TextureUsageBit::DEPTH_STENCIL_ATTACHMENT,
+            gfx::Format::DEPTH_STENCIL,
+            static_cast<uint>(_pipeline->getWidth() * shadingScale),
+            static_cast<uint>(_pipeline->getHeight() * shadingScale),
+        };
+        framegraph::RenderTargetAttachment::Descriptor depthAttachmentInfo;
+        depthAttachmentInfo.usage = framegraph::RenderTargetAttachment::Usage::DEPTH_STENCIL;
+        depthAttachmentInfo.loadOp = gfx::LoadOp::CLEAR;
+        depthAttachmentInfo.clearDepth = camera->getClearDepth();
+        depthAttachmentInfo.clearStencil = camera->getClearStencil();
+        depthAttachmentInfo.endAccesses = {gfx::AccessFlagBit::DEPTH_STENCIL_ATTACHMENT_WRITE};
+
+        data.depth = builder.create(RenderPipeline::fgStrHandleOutDepthTexture, depthTexInfo);
+        data.depth = builder.write(data.depth, depthAttachmentInfo);
+        
+        builder.writeToBlackboard(RenderPipeline::fgStrHandleOutColorTexture, data.outputTex);
+        builder.writeToBlackboard(RenderPipeline::fgStrHandleOutDepthTexture, data.depth);
+        builder.setViewport(_pipeline->getViewport(camera), _pipeline->getScissor(camera));
+    };
+
+    auto offset = _pipeline->getPipelineUBO()->getCurrentCameraUBOOffset();
+    auto arExec = [this, camera, offset](const RenderData & /*data*/, const framegraph::DevicePassResourceTable &table) {
+        auto *cmdBuff = _pipeline->getCommandBuffers()[0];
+
+        auto *renderPass = table.getRenderPass();
+        _arBackground->render(camera, renderPass, cmdBuff);
+    };
+
+    // add pass
+    _pipeline->getFrameGraph().addPass<RenderData>(static_cast<uint>(CommonInsertPoint::DIP_AR_BACKGROUND), ForwardPipeline::fgStrHandleForwardPass, arSetup, arExec);
```

Comment:
Maybe a fullscreen quad like skybox is more efficient than a new pass?

Reply:
Maybe compatibility can be considered later